geopandas/plotting.py

```diff
@@ -36,12 +41,15 @@ def _flatten_multi_geoms(geoms, colors):
         else:
             components.append(geom)
             component_colors.append(color)
+
+    if was_none:
+        component_colors = None
```

Comment:
Since you assign `colors` in line 29, then `component_colors` should already be a list of `None` that matches the length of `geoms`.  I'd find the code clearer if  you remove `was_none` from this function, and instead do a check for `None` in the calling code _after_ `_flatten_multi_geoms` returns.  It's a matter of separation of concerns, since it is the calling code that wants to do further special-casing if `values is None`: 

    geoms, values = _flatten_mutli_geoms(geoms, values)
    if None in values:
        values = None

In any case, though, you are breaking the function's return signature (goes against docstring).